### PR TITLE
update rust to 10-10-2024

### PR DESCRIPTION
**Pull request by aviramha**
Please do a through review.

### CODE REVIEW

**Review by Razz4780 — DISMISSED** [review]

**Comment by aviramha** [comment]
can't do it yet ;( @meowjesty was right

**Comment by aviramha** [comment]
ok turns out @meowjesty wasn't right, the error would never be gone because the code that caused it is evil and needs to be refactored!
luckily, I have entered the belly of the beast and took the beast's heart out!

**Comment by aviramha** [comment]
We should merge this after next release.

File: CHANGELOG.md [hunk 1]
```diff
@@ -949,7 +949,7 @@ This project uses [*towncrier*](https://towncrier.readthedocs.io/) and the chang
   onto `KubernetesAPI`.
 - Split off pod template to a separate `ContainerVariant` implementation used
   inside of `JobVariant` and `JobTargetedVariant`.
-- Updates to rust nightly-2024-04-15. Also fixes some pointer
+- Updates to rust nightly-2024-10-11. Also fixes some pointer
```
Comment (DmitryDodzin) [inline CHANGELOG.md:952]: I don't think this version is needed to be updated

Comment (aviramha) [reply CHANGELOG.md:952]: nice catch!

**Review by DmitryDodzin — APPROVED** [review]
LGTM

**Pull request merged**